index.js

```diff
@@ -94,39 +93,30 @@ module.exports = class Hyperdrive extends ReadyResource {
     return new Hyperdrive(this.corestore, this.key, {
       onwait: this._onwait,
       _checkout: null,
-      _db: this.db,
-      _files: this.files.batch()
+      _db: this.db.batch()
     })
   }
 
-  async flush () {
-    await this.files.flush()
-    return this.close()
+  flush () {
+    return this.db.flush()
```

Comment:
Sanity checking: we no longer need to close this upon flushing a batch? Not really understanding why it's no longer needed, as `Hyperdrive.batch()` does return a new `Hyperdrive` object

Also sanity checking: it's normal that `batch()` returns a Hyperdrive object on the same `this.corestore` instance, without creating a new session? That seems wrong (definitely if we close the passed in `Hyperdrive` on flushing, as I think we should)

Reply:
Just noticed that we conditionally close the corestore in `Hyperdrive.close()`, not closing it if we're in a checkout or batch, so calling close with the code as-is seems fine, it won't close the corestore in a batch.

Do note that this means that, when we close a Hyperdrive, any existing checkouts will become unusable (because the underlying corestore is closed), but that might be expected.

Reply:
I think you're right! The reason I didn't close it is because in main branch I didn't immediately see `this.files` being closed but it's actually in the close method so my bad!

Also, it should close it because MirrorDrive expects it as well:
https://github.com/holepunchto/mirror-drive/blob/cbd6dea2f175522536cb767f38fc6d23c9712677/index.js#L87
I mean, it does flush but it doesn't close, I think Mathias saw that so flush must close